sdk/core/wasm/src/merkle.rs

```diff
@@ -0,0 +1,181 @@
+use wasm_bindgen::prelude::*;
+
+use parity_scale_codec::{Compact, Decode, Encode};
+use shp_constants::FILE_CHUNK_SIZE;
+use shp_file_metadata::{Chunk, ChunkId, ChunkWithId};
+use sp_core::{hashing::blake2_256, Hasher};
+use sp_trie::{LayoutV1, MemoryDB, TrieDBMutBuilder, TrieMut};
+use std::collections::hash_map::DefaultHasher;
+
+// ────────────────────────────────────────────────────────────────────────────
+//  NOTE:
+//  The full StorageHub client crate (`client/file-manager`) depends on a large
+//  graph of Substrate-runtime crates that pull in features which cannot be
+//  compiled to `wasm32-unknown-unknown` (e.g. `std` I/O, rocksdb, proc-macro
+//  helpers, etc.).
+//
+//  For the SDK we only need the minimal in-memory Merkle-Patricia trie logic
+//  (`InMemoryFileDataTrie`) to compute Merkle roots client-side.  Therefore we
+//  copy the few small types (ChunkId, ChunkWithId, hashing adapter) instead of
+//  depending on the heavy client crate.  This keeps the WASM package tiny and
+//  avoids compilation failures in browser/Node environments.
+// ────────────────────────────────────────────────────────────────────────────
```

Comment:
I like the approach (keeping the wasm small), but ideally we would move out these items in a separate crate to depend on instead: we avoid duplication and get a semi-sdk in rust. 
Up to you if you want to do that at a later point, but I think sooner is better

Reply:
I agree with @Karrq 

But I believe we can move perhaps the types to one of the common crates we have if it does not depend on any std deps

Reply:
I see, I think it is reasonable to add a TODO to refactor this later than